### PR TITLE
Migrate dnsmasq config files

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Migrate `dnsmasq` config files originally created by a Pi-hole version pre-v6.0 from `/etc/dnsmasq.d` into a new directory owned by user/group `pihole`. This ensures they stay available for later debugging, and won't cause when, e.g., the user later decides to install a `dnsmasq` from the repository.

The differences to how it is done before are:
1. The files are moved completely out of the way of any possibly external `dnsmasq` and won't cause conflicts,
2. The files can be moved by the installer even if "something" happens in between and the files are now owned by, e.g., `root` (`pihole-FTL` could then not move/rename them itself).

This PR is meant to be used in combination with https://github.com/pi-hole/FTL/pull/1737

---

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
3. I have commented my proposed changes within the code.
4. I am willing to help maintain this change if there are issues with it later.
5. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
6. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -2070,6 +2073,34 @@ copy_to_install_log() {
...
+
+    # Exit early if this is already Pi-hole v6.0
+    # We decide this on the presence of the file /etc/pihole/pihole.toml
+    if [[ -f /etc/pihole/pihole.toml ]]; then
+        return 0
+    fi
+
+    # Create target directory /etc/pihole/migration_backup_v6
+    # and make it owned by pihole:pihole
+    mkdir -p "${V6_CONF_MIGRATION_DIR}
```
Comment (PromoFaux) [inline automated install/basic-install.sh:2090]:
```suggestion
    mkdir -p "${V6_CONF_MIGRATION_DIR}"
```

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -2070,6 +2073,34 @@ copy_to_install_log() {
...
+    chown pihole:pihole "${V6_CONF_MIGRATION_DIR}
```
Comment (PromoFaux) [inline automated install/basic-install.sh:2091]:
```suggestion
    chown pihole:pihole "${V6_CONF_MIGRATION_DIR}"
```

File: automated install/basic-install.sh [hunk 3]
```diff
@@ -2070,6 +2073,34 @@ copy_to_install_log() {
...
+
+    # Move all conf files originally created by Pi-hole into this directory
+    # - 01-pihole.conf
+    # - 02-pihole-dhcp.conf
+    # - 04-pihole-static-dhcp.conf
+    # - 05-pihole-custom-cname.conf
+    # - 06-rfc6761.conf
+
+    mv /etc/dnsmasq.d/0{1,2,4,5}-pihole*.conf "${V6_MIGRATION_DIR}/ CONF_2>/dev/null || true
```
Comment (PromoFaux) [inline automated install/basic-install.sh:2100]:
```suggestion
    mv /etc/dnsmasq.d/0{1,2,4,5}-pihole*.conf "${V6_MIGRATION_DIR}"/ CONF_2>/dev/null || true
```

File: automated install/basic-install.sh [hunk 4]
```diff
@@ -2070,6 +2073,34 @@ copy_to_install_log() {
...
+    mv /etc/dnsmasq.d/06-rfc6761.conf "${V6_MIGRATION_DIR}/ CONF_2>/dev/null || true
```
Comment (PromoFaux) [inline automated install/basic-install.sh:2101]:
```suggestion
    mv /etc/dnsmasq.d/06-rfc6761.conf "${V6_MIGRATION_DIR}"/ CONF_2>/dev/null || true
```

**Review by PromoFaux — CHANGES_REQUESTED** [review]
Get some rest!

File: automated install/basic-install.sh (hunk 3 above)
Comment (PromoFaux) [inline automated install/basic-install.sh:2100]:
```suggestion
    mv /etc/dnsmasq.d/0{1,2,4,5}-pihole*.conf "${V6_MIGRATION_DIR}/" CONF_2>/dev/null || true
```

File: automated install/basic-install.sh (hunk 4 above)
Comment (PromoFaux) [inline automated install/basic-install.sh:2101]:
```suggestion
    mv /etc/dnsmasq.d/06-rfc6761.conf "${V6_MIGRATION_DIR}/" CONF_2>/dev/null || true
```

File: automated install/basic-install.sh (hunk 3 above)
Comment (PromoFaux) [reply automated install/basic-install.sh:2100]: Actually, no... same for the line below. What exactly are you trying to do with `CONF_2` in this case?

Comment (rdwebdesign) [reply automated install/basic-install.sh:2100]: Apparently this is a typo.
There is no variable called `V6_MIGRATION_DIR`.

Maybe the code should be something like this:

```suggestion
    mv /etc/dnsmasq.d/0{1,2,4,5}-pihole*.conf "${V6_CONF_MIGRATION_DIR}"/ 2>/dev/null || true
```

File: automated install/basic-install.sh [hunk 5]
```diff
@@ -2070,6 +2073,34 @@ copy_to_install_log() {
...
+
+    # Move all conf files originally created by Pi-hole into this directory
+    # - 01-pihole.conf
+    # - 02-pihole-dhcp.conf
+    # - 04-pihole-static-dhcp.conf
+    # - 05-pihole-custom-cname.conf
+    # - 06-rfc6761.conf
+
+    mv /etc/dnsmasq.d/0{1,2,4,5}-pihole*.conf "${V6_CONF_MIGRATION_DIR}/" 2>/dev/null || true
+    mv /etc/dnsmasq.d/06-rfc6761.conf "${V6_MIGRATION_DIR}/" 2>/dev/null || true
```
Comment (DL6ER) [inline automated install/basic-install.sh:2101]:
```suggestion
    mv /etc/dnsmasq.d/06-rfc6761.conf "${V6_CONF_MIGRATION_DIR}/" 2>/dev/null || true
```

**Comment by DL6ER** [comment]
Sorry, I don't really know what happened here. The file somehow got mangled after I tested it... Thanks for your thorough check!

**Comment by PromoFaux** [comment]
Does this branch need rebasing? Just built a local container based on it and:

```
++ pihole-FTL sqlite3 -ni /etc/pihole/gravity.db 'SELECT "value" FROM "info" WHERE "property" = '\''version'\'';'
pihole  | sqlite3: Error: unknown option: -ni
pihole  | Use -help for a list of options.
pihole  | + version=
```

**Comment by DL6ER** [comment]
Indeed, a `merge` did just happen

**Comment by DL6ER** [comment]
This PR still needs merging to ensure we remove files from `/etc/dnsmasq.d` Pi-hole has put there before

**Review by rdwebdesign — APPROVED** [review]

**Pull request merged**